iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingViewIndex.java

```diff
@@ -74,21 +76,24 @@ void checkAndAdd(final Network other) {
         other.getDanglingLineStream().forEach(this::checkNewDanglingLine);
     }
 
-    void checkNewDanglingLine(final DanglingLine dll2) {
-        Objects.requireNonNull(dll2, "DanglingLine is null");
-        // Manage DanglingLines
-        final String code = dll2.getUcteXnodeCode();
-        if (code == null) {
-            return;
-        }
-        // Find other DanglingLine if exist
-        final Set<DanglingLine> danglingLines = getNetworkStream().flatMap(Network::getDanglingLineStream).filter(d -> d.getUcteXnodeCode().equals(code)).collect(Collectors.toSet());
-        for (DanglingLine dll1 : danglingLines) {
-            if (dll1 != dll2) {
-                // Create MergedLine from 2 dangling lines
-                mergedLineCached.computeIfAbsent(code, key -> new MergedLine(this, dll1, dll2));
-            }
-        }
+    void checkNewDanglingLine(DanglingLine dl2) {
+        checkAssociatedDanglingLines(dl2, id -> getNetworkStream().map(n -> n.getDanglingLine(id))
+                .filter(Objects::nonNull)
+                .filter(dl -> dl != dl2)
+                .findFirst()
+                .orElse(null), code -> getNetworkStream()
+                .flatMap(Network::getDanglingLineStream)
+                .filter(d -> code.equals(d.getUcteXnodeCode()))
+                .filter(d -> d != dl2)
+                .filter(d -> d.getNetwork() != dl2.getNetwork())
+                .distinct()
+                .collect(Collectors.toList()), (dll1, dll2) -> {
+                String key = dll1.getId();
+                if (dll1.getUcteXnodeCode() != null && dll2.getUcteXnodeCode() != null) {
+                    key = dll1.getUcteXnodeCode();
+                }
+                mergedLineCached.computeIfAbsent(key, k -> new MergedLine(this, dll1, dll2));
+            });
```

Comment:
I was a bit lost in the arguments here, could you define them on separate lines?